knowage-core/src/main/java/it/eng/spagobi/analiticalmodel/document/service/ManagePreviewFileAction.java

```diff
@@ -107,6 +109,11 @@ public void doService() {
 	// checks for path traversal attacks
 	private void checkRequiredFile(String fileName) {
 		File targetDirectory = GeneralUtilities.getPreviewFilesStorageDirectoryPath();
+		try {
+			PathTraversalChecker.get(fileName, targetDirectory.getName());
+		} catch (Exception e) {
+			throw new PathTraversalAttackException("Error managing preview file for file: " + fileName);
+		}
```

Comment:
PathTraversalChecker should throw only PathTraversalAttackException exception, that is runtime exception, so we can remove the try catch bloch